mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+
+    // Creating a boolean variable to tell us when help us requested.
+    bool helpRequested{false};
+
+    // Creating a boolean variable to tell me when the version is requested.
+    bool versionRequested{false};
+
+    // Creating a boolean to check if the input file was given and initialising the input file variable
+    bool input_fileGiven{false};
+    std::string input_file{""};
+
+    // Creating a boolean to check if the output file was given and initialising the output file variable
+    bool output_fileGiven{false};
+    std::string output_file{""};
+
+    // Looking through the command line arguments, excluding the first one.
+    for (std::size_t i{1}; i < cmdLineArgs.size(); i++){
+
+        //std::cout <<  " The argument being considered now is " << cmdLineArgs[i] << std::endl;
+        //std::cout <<  "The argument index is " << i << std::endl;
+
+        //Looking for whether help is requested
+        if((cmdLineArgs[i] == "--help") || (cmdLineArgs[i] == "-h") ){
+
+            helpRequested = true;
+
+        }
+
+        // Looking for whether the version is requested
+        if(cmdLineArgs[i] == "--version" ){
+
+            versionRequested = true;
+
+        }
+
+
+        // Looking for whether the option -i has been used as an argument
+        if(cmdLineArgs[i] == "-i" ){
+            
+            // Need to look to see if this was the last argument specified
+            // If it was, need to return an error message.
+            if (i == (cmdLineArgs.size() - 1)){
+                std::cout << "An input file must be specified in the -i option after '-i'." << std::endl;
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+
+                // Take the next argument as the input file
+                input_fileGiven = true;
+                //std::cout << input_fileGiven << std::endl;
+                input_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an input file is " << input_file << std::endl;
+
+            }
+
+            
+        }
+        
+
+
+
+        // Looking for whether the option -o has been used as an argument
+        if(cmdLineArgs[i] == "-o" ){
+
+            // Need to look to see if this was the last argument specified
+            // If it was, need to return an error message.
+            if (i == (cmdLineArgs.size() - 1)){
+                std::cout << "An output file must be specified in the -o option after '-o'." << std::endl;
+                break;
+            }
+            else{
+                // If it wasn't, proceed as normal.
+            
+                // Take the next argument as the output file
+                output_fileGiven = true;
+                output_file = cmdLineArgs[i+1];
+                //std::cout << "The file identified as an output file is " << output_file << std::endl;
+            }
+           
+        }
+        
```

Comment:
You should have an `else` at the end to catch unexpected arguments, in which case you should print a corresponding error message and quit.